sympy/polys/polytools.py

```diff
@@ -5611,10 +5611,18 @@ def _symbolic_factor_list(expr, opt, method):
     """Helper function for :func:`_symbolic_factor`. """
     coeff, factors = S.One, []
 
-    for arg in Mul.make_args(expr):
+    args = list(Mul.make_args(expr))
+    for arg in args:
         if arg.is_Number:
             coeff *= arg
             continue
+        elif hasattr(arg, '_eval_factor'):
+            for i in Mul.make_args(arg._eval_factor()):
```

Comment:
I'm pretty sure you don't want to do it like this. If you have an unsuccessful factoring, you are putting the result on the stack and will end up with an infinite loop. This is the full rendition of what I intended:


```python
def _symbolic_factor_list(expr, opt, method):
    """Helper function for :func:`_symbolic_factor`. """
    coeff, factors = S.One, []

    args = list(Mul.make_args(expr))
    for arg in args:
        if arg.is_Number:
            coeff *= arg
            continue
        if arg.is_Pow:
            base, exp = arg.args
            if base.is_Number:
                factors.append((base, exp))
                continue
        if hasattr(arg, '_eval_factor'):
            arg = arg._eval_factor()
            if arg.is_Mul:
                args.extend(arg.args)
                continue
        base, exp = arg, S.One

        try:
            ...
```

The reason I do it like that is there is some pretty non-trivial code that happens to each args: by just putting factors on the `args` stack you let them be processed consistently without duplicating efforts (other than refactoring an object with the _eval_factor method unsuccessfully the 2nd pass through).